src/sqlfluff/rules/L016.py

```diff
@@ -429,14 +428,25 @@ def _eval(self, segment, raw_stack, **kwargs):
                         idx -= 1
                     else:
                         break  # pragma: no cover
+                create_elements = line_indent + [this_line[-1], segment]
+                if sum(len(s.raw) for s in create_elements) > self.max_line_length:
+                    # The inline comment is NOT on a line by itself, but even if
+                    # we move it onto a line by itself, it's still too long. In
+                    # this case, the rule should do nothing, otherwise it
+                    # triggers an endless cycle of "fixes" that simply keeps
+                    # adding blank lines.
+                    self.logger.info(
```

Comment:
Reviewers: The lines inside the `if` are very similar to existing lines 407-413, only the condition is different.